validation/build.gradle

```diff
@@ -40,8 +40,13 @@ dependencies {
 
     implementation "org.apache.flink:flink-scala_${scalaBinaryVersion}:${flinkVersion}"
     implementation "org.apache.flink:flink-streaming-scala_${scalaBinaryVersion}:${flinkVersion}"
-    implementation "log4j:log4j:${log4jVersion}"
-    implementation "org.slf4j:slf4j-log4j12:${slf4jVersion}"
+    implementation "org.apache.logging.log4j:log4j-core:${log4jVersion}"
+    implementation "org.apache.logging.log4j:log4j-slf4j-impl:${log4jVersion}"
+    compileOnly("org.scala-lang:scala-compiler") {
```

Comment:
Does this work since the flink-scala stuff is an implementation dependency? In passthrough which is similar if I made it compileOnly I still see the vulnerability. I had to make the scala-compiler an implementation level dependency, and then exclude it from the shadowJar.